nautobot/docs/user-guides/git-data-source.md

```diff
@@ -0,0 +1,257 @@
+
+Filename: `/export_templates/dcim/device/yaml_export.yml`
+
+Current contents:
+```jinja
+---
+{% for device in queryset %}
+{% if device.status %}
+- {{ device.name }}:
+{% endif %}
+{% endfor %}
+...
+```
+
+The template needs to be modified to provide more information than just a list of hostnames.  The site needs to be added.
+
+The updated template is now:
+```jinja
+---
+{% for device in queryset %}
+{% if device.status %}
+- {{ device.name }}:
+  site: {{ device.site }}
+{% endif %}
+{% endfor %}
+...
+```
+
+Once the contributor updates the Git repository via normal Git processes and it is reviewed and merged into the branch that was used, a sync process from Nautobot needs to be completed.  This can be done from the default Git view, or within a specific detailed view of a Git repository.
+
+From the default Git repositories view:
+![](./images/git-data-source_8.png)
+
+From the detailed view:
+![](./images/git-data-source_9.png)
+
+> Once the repository has been synced its easy to check the history for the templates.
+Navigate to **Git Repositories** and select the repository in question.  Once you're in the detailed view you can look through the **Synchronization Status** or **Change Log** tabs.
+
+Now that the Git repository is linked for export templates it can be controlled via the normal Git operations workflow, which allows users or groups of users to perform code reviews using Pull Requests etc.
+
+### Jobs
+Jobs are a way for users to execute custom logic on demand from within the Nautobot UI. Jobs can interact directly with Nautobot data to accomplish various data creation, modification, and validation tasks.
+
+For technical details on jobs, please see the documentation on [jobs](https://nautobot.readthedocs.io/en/latest/additional-features/jobs/#jobs).
+
+Jobs allow a user to write scripts in Python.  By integrating the scripts with Git, a user can utilize Git workflows to manage source control, versioning, and pipelines.
+
+Setting up the repository can be done following the same steps from [Export Templates](#Export-Templates).  The only differences is the `provides` selection changes to `jobs`.
+
+Jobs need to be defined in `/jobs/` directory at the root of a Git repository.
+
+An example tree for `/jobs/`.
+
+```
+▶ tree jobs 
+jobs
+├── __init__.py
+└── get-device-connection.py
+
+1 directory, 2 files
+```
+
+Once the repository is created in Nautobot.
+![](./images/git-data-source_10.png)
+
+> Note: The same repository and branch can be used for the different `provides` methods.  Nautobot Git as a data source will look for specific root directory names.
+
+Once the scripts have been pushed into the repository, a sync needs to be executed, after which navigating to Jobs via **Extensibility -> Jobs** will show the new jobs loaded from the Git repository.
+
+![](./images/git-data-source_11.png)
+
+Jobs now shows the job from the Git repository.
+
+![](./images/git-data-source_12.png)
```

Comment:
Same comment about this screenshot, as well as images 4 through 9 and 14 through 18.